### ISSUE TITLE
No sound on iOS in production

### ISSUE THREAD

**Issue by koravski** [expo/expo#13558]
### Summary

Hello, I have a simple app that uses sounds from mp3 files. When clicking on a button the sounds play once. I was on SDK 39 and it was working fine. Then updated to SDK 42 and I was thinking it was fine. The sound is working ok in development but once I publish it there is no sound. The problem is only on iOS. Tested with SDK 41 and the same problem. I’m using expo-av latest version with SDK 42. Tested with a generated build for simulator as well, the same problem.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

42

### Environment

Expo CLI 4.7.3 environment info:
    System:
      OS: macOS 11.4
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.5.0 - ~/.nvm/versions/node/v14.5.0/bin/node
      Yarn: 1.22.10 - /usr/local/bin/yarn
      npm: 7.16.0 - ~/.nvm/versions/node/v14.5.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    IDEs:
      Android Studio: 4.2 AI-202.7660.26.42.7322048
      Xcode: 12.5/12E262 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^42.0.0 => 42.0.0 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: https://github.com/expo/react-native/archive/sdk-42.0.0.tar.gz => 0.63.2 
      react-native-web: ~0.13.12 => 0.13.18 
    npmGlobalPackages:
      expo-cli: 4.7.3
    Expo Workflow: managed

### Reproducible demo or steps to reproduce from a blank project

Tested with a brand new app with expo init.
Added the code from the documentation. 
Working fine in development, when published no sound on iOS.

```
import * as React from 'react';
import { Text, View, StyleSheet, Button } from 'react-native';
import { Audio } from 'expo-av';

export default function App() {
  const [sound, setSound] = React.useState();

  async function playSound() {
    console.log('Loading Sound');
    const { sound } = await Audio.Sound.createAsync(
       require('./assets/01.mp3')
    );
    setSound(sound);

    console.log('Playing Sound');
    await sound.playAsync(); }

  React.useEffect(() => {
    return sound
      ? () => {
          console.log('Unloading Sound');
          sound.unloadAsync(); }
      : undefined;
  }, [sound]);

  return (
    <View>
      <Button title="Play Sound" onPress={playSound} />
    </View>
  );
}
```

**Comment by u840903** [comment]
Since yesterday I'm having problem with playing audio as well. I'm loading audio with...
```ts
 const {sound} = await Audio.Sound.createAsync({uri})
 ```
 ...and since yesterday it always works the first time, but when I change uri I sometimes get.
```
[Unhandled promise rejection: Error: g.g.b.d.t0.v$b: okhttp3.internal.http2.StreamResetException: stream was reset: PROTOCOL_ERROR]
at node_modules/react-native/Libraries/BatchedBridge/NativeModules.js:103:50 in promiseMethodWrapper
at node_modules/@unimodules/react-native-adapter/build/NativeModulesProxy.native.js:15:23 in moduleName.methodInfo.name
at node_modules/expo-av/build/Audio/Sound.js:55:4 in _performOperationAndHandleStatusAsync
at node_modules/expo-av/build/Audio/Sound.js:55:4 in _performOperationAndHandleStatusAsync
at [native code]:null in flushedQueue
at [native code]:null in callFunctionReturnFlushedQueue
```
The same file sometimes works, sometimes it does not.

**Comment by koravski** [comment]
I think it's from the expo go (2.21.3) app or SDK 42. 
Now I reverted back to SDK 41. There is no sound in the latest expo app on a physical device, but when I built the app there is sound. Also if I go back to the previous expo app version on the simulator there is sound again.

**Comment by jwoo92** [comment]
I am experiencing something similar as well.

Using bare workflow with [expo custom development client](https://docs.expo.io/clients/getting-started/). works fine in dev + production mode of custom development client, but audio appears to not play in the stand-alone release build.

---

**physical devices tested**:

* iPhone 12 Pro, iOS 14.6
* iPad mini (5th generation), iPadOS 14.6

**related packages**:

* react-native: `0.63.4`
* expo: `42.0.0`
* expo-av: `9.2.3`
* expo-dev-client: `0.4.3`
* @unimodules/core: `7.1.1`
* react-native-unimodules: `0.14.5`

**Comment by c-mcdonald** [comment]
I'm seeing the same behaviour on iOS. Sound worked fine for Expo 41 with expo-av. Upgraded to Expo 42, sound works fine in development iOS and Android builds, but store binaries for iOS, no sound. Android aab is fine.

Bumped expo-av back to 9.1.2 and added the supporting filesystem and permissions unimodules, still no sound.

Only iOS and only store builds.

**Comment by brentvatne** [comment]
the only meaningful change to expo-av on ios in SDK 42 was this: https://github.com/expo/expo/commit/1f6de6b9a85c27c5aa1f992fa405c52e4c41e652#diff-f71d6df6a9bcca5e253da4998772ff904d9c8b446017d6c132f1307600600ef9

you can refer to the commit log here: https://github.com/expo/expo/commits/master/packages/expo-av

you can try reverting the aforementioned commit locally in your node_modules to see if that will fix the issue for you

let me know what you find, and please create a minimal reproducible example project on github and share it here

**Comment by hugotox** [comment]
I have the same issue: audio works fine in the simulator and real iOS device in dev mode. But the production build doesn't have sound in the real device. 
FWIW I haven't change anything in my package.json, relevant lines:
```
    "expo": "~41.0.1",
    "expo-app-loading": "^1.0.3",
    "expo-av": "~9.1.2",
    "expo-font": "~9.1.0",
    "expo-keep-awake": "~9.1.2",
    "expo-linear-gradient": "^9.1.0",
    "expo-status-bar": "~1.0.4",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-native": "https://github.com/expo/react-native/archive/sdk-40.0.1.tar.gz",
```
So I suspect the problem is with the expo client app in iOS, which was the only thing updated lately

**Comment by stuckj** [comment]
Check your ringer / silent switch on your iOS device. By default, video won't override the ringer / silent switch. You can override it per the [docs](https://docs.expo.io/versions/latest/sdk/audio/#audiosetaudiomodeasyncmode), but I just noticed a [bug](https://github.com/expo/expo/issues/13617) where overriding the silent switch doesn't appear to work if you don't autoplay the video (just submitted it). Not sure if that's the issue, but figured I'd mention it as a thing to check. :)

**Comment by reinisindans** [comment]
Same problem here: 
 No sound on iOS in production app, when trying the app through Expo Go App after publishing it to Expo. 
 
Tried reverting the changes to axpo-av as suggested by @brentvatne , but to no avail. 

Tend to agree with @hugotox , about Expo Go app on iOS being the real culprit here. 
Hopefully will resolve fast, have to share my app with team.

**Comment by afahim12** [comment]
Same. Audio doesn’t play nor does animation.

**Comment by c-mcdonald** [comment]
Thanks @brentvatne I tried reverting the mentioned changes in expo-av but didn't make a difference.

I reverted my project back to SDK 41 and ran a standalone build for iOS using EAS, confirmed audio works. Ran Expo upgrade to SDK 42, no other changes, ran a standalone build for iOS using EAS, audio no longer plays.

As I am in bare workflow, I don't think this is an issue isolated to the Expo client app.

Seems like there have been a lot of plumbing changes between Expo 41 and 42 - thinks like refactoring permissions, filesystem constants modules. Could there be something underlying there that changed?

Just to clarify this is only on standalone iOS builds. Android and iOS simulator builds play audio just fine.

**Comment by hugotox** [comment]
Definitely is something with the device, maybe the latest iOS update. I just installed a previous version of my app that I'm 100% sure had sound, but now it doesn't.

**Comment by TMcLoone** [comment]
I don't believe this is an expo-av issue but definitely something in expo 42. I removed expo-av and only using react-native-video and still experiencing no sound in production but works on the simulator. I reverted back to 41 and audio returns without issue.

**Comment by brentvatne** [comment]
@TMcLoone - are you using expo-updates in your app?

**Comment by TMcLoone** [comment]
> @TMcLoone - are you using expo-updates in your app?
> 
> 
> 
> can someone share a [mcve](https://stackoverflow.com/help/minimal-reproducible-example)?

Yes using expo-updates. Can't get to a MCVE today sorry

**Comment by hugotox** [comment]
@brentvatne sure, this is actually pretty straightforward 

https://github.com/hugotox/expo-audio-example

Published here:

https://expo.io/@hugotox/expo-audio-example

**Comment by brentvatne** [comment]
thanks, we'll look into it tomorrow

**Comment by brentvatne** [comment]
this is resolved in expo-asset@8.3.3. update to expo@42.0.3 and/or react-native-unimodules@0.14.6 to have this new expo-asset release pulled in automatically. run `yarn why expo-asset` after updating those packages to ensure the expo-asset version is 8.3.3 or later (or the equivalent for npm, not sure what that is -- you could check package-lock.json).

this regression was caused by https://github.com/expo/expo/commit/8672a2af097fe469c3b256ac930efb91b4a2f528, which we were able to safely revert because it doesn't impact any public facing features (except, accidentally, this)
